third_party/amd/lib/TritonAMDGPUTransforms/MfmaGroup.cpp

```diff
@@ -258,12 +340,26 @@ std::pair<mlir::Type, mlir::Type> TypesFromMfmaId(mlir::MLIRContext *ctx,
 }
 
 FailureOr<MfmaInsn> MfmaInsn::selectMfma(unsigned mDim, unsigned nDim,
-                                         Type elementTypeA, Type elementTypeB,
-                                         int mfmaVersion, bool allowXF32) {
+                                         unsigned kDim, Type elementTypeA,
+                                         Type elementTypeB, int mfmaVersion,
+                                         bool allowXF32) {
   auto mfmaInsnAttrMap = getMfmaInsnGroupAttrMap();
   MfmaTypeId mfmaId =
       chooseAppropriateMfmaId(elementTypeA, elementTypeB, allowXF32);
-  MfmaInsnGroupSelectKey key = {mDim, nDim, mfmaId, mfmaVersion};
+  // gfx950: select double-rate mfma ops for large kDim
+  if (mfmaVersion == 4 &&
```

Comment:
See above about swap this out for the GPUKind (e.g. llvm::AMDGPU::GK_GFX950). The MFMA instructions available should be tied to the GPU arch.

Reply:
This is fine for now--mfma version is used throughout the codebase. If we have other ideas to represent, it's better to happen separately.